File: FVDocument.m
//
//  FVDocument.m
//  ForkView
//
//  Created by Kevin Wojniak on 5/25/11.
//  Copyright 2011 Kevin Wojniak. All rights reserved.
//

#import "FVDocument.h"
#import "FVWindowController.h"
#import "FVResourceFile.h"
#import "FVResource.h"

@implementation FVDocument

@synthesize resourceFile;

- (void)dealloc
{
	[windowController release];
	[resourceFile release];
	[super dealloc];
}

- (void)makeWindowControllers
{
	[super makeWindowControllers];
	
	windowController = [[FVWindowController alloc] init];
	[self addWindowController:windowController];
}

- (BOOL)readFromURL:(NSURL *)absoluteURL ofType:(NSString *)typeName error:(NSError **)outError
{
	self.resourceFile = [FVResourceFile resourceFileWithContentsOfURL:absoluteURL error:outError];
	if (self.resourceFile == nil) {
		return NO;
	}
	return YES;
}

@end

File: FVDocument.swift
//
//  FVDocument.swift
//  ForkView
//
//  Created by Kevin Wojniak on 8/3/14.
//  Copyright (c) 2014 Kevin Wojniak. All rights reserved.
//

import Cocoa

@NSApplicationMain final class FVDocument: NSDocument, NSApplicationDelegate {
    @objc var resourceFile: FVResourceFile?
    var windowController: NSWindowController?

    override func makeWindowControllers() {
        super.makeWindowControllers()

        windowController = FVWindowController.windowController()
        addWindowController(windowController!)
    }

    override func read(from url: URL, ofType typeName: String) throws {
        resourceFile = try FVResourceFile.resourceFileWithContentsOfURL(fileURL: url)
    }
}
